spark/src/main/scala/geotrellis/spark/io/accumulo/spacetime/SpaceTimeRasterRDDReaderProvider.scala

```diff
@@ -0,0 +1,144 @@
+  import SpaceTimeRasterRDDIndex._
+
+  def tileSlugs(filters: List[GridBounds]): List[(String, String)] = filters match {
+    case Nil =>
+      List(("0"*6 + "_" + "0"*6) -> ("9"*6 + "_" + "9"*6))
+    case _ => 
+      for{
+        bounds <- filters
+        row <- bounds.rowMin to bounds.rowMax 
+      } yield f"${bounds.colMin}%06d_${row}%06d" -> f"${bounds.colMax}%06d_${row}%06d"      
+  }
+  
+  def timeSlugs(filters: List[(DateTime, DateTime)], minTime: DateTime, maxTime: DateTime): List[(Int, Int)] = filters match {
+    case Nil =>
+      List(timeChunk(minTime).toInt -> timeChunk(maxTime).toInt)
+    case List((start, end)) =>                 
+      List(timeChunk(start).toInt -> timeChunk(end).toInt)
+  }
+
+  def setFilters(job: Job, layerId: LayerId, filterSet: FilterSet[SpaceTimeKey], keyBounds: KeyBounds[SpaceTimeKey]): Unit = {
+    var spaceFilters: List[GridBounds] = Nil
+    var timeFilters: List[(DateTime, DateTime)] = Nil
+
+    filterSet.filters.foreach {
+      case SpaceFilter(bounds) => 
+        spaceFilters = bounds :: spaceFilters
+      case TimeFilter(start, end) =>
+        timeFilters = (start, end) :: timeFilters
+    }
+
+    def timeIndex(dt: DateTime) = timeChunk(dt).toInt
+
+    InputFormatBase.setLogLevel(job, org.apache.log4j.Level.DEBUG)
+    
+    val ranges: Seq[ARange] = (
+      for {
+        bounds <- spaceFilters
+        (timeStart, timeEnd) <- timeSlugs(timeFilters, keyBounds.minKey.temporalKey.time, keyBounds.maxKey.temporalKey.time)
+      } yield {
+        val p1 = Z3(bounds.colMin, bounds.rowMin, timeStart)
+        val p2 = Z3(bounds.colMax, bounds.rowMax, timeEnd)
+        
+        val rangeProps = Map(
+          "min" -> p1.z.toString,
+          "max" -> p2.z.toString)
+        
+
+        val ranges = Z3.zranges(p1, p2)
+
+        ranges
+          .map { case (min: Long, max: Long) =>
+
+            val start = f"${layerId.zoom}%02d_${min}%019d"
+            val end   = f"${layerId.zoom}%02d_${max}%019d"
+            val zmin = new Z3(min)
+            val zmax = new Z3(max)      
+            if (min == max)
+              ARange.exact(start)
+            else
+              new ARange(start, true, end, true)
+          }        
+      }).flatten
+
+    InputFormatBase.setRanges(job, ranges)
+
+    for ( (start, end) <- timeFilters) {
+      val props =  Map(
+        "startBound" -> start.toString,
+        "endBound" -> end.toString,
+        "startInclusive" -> "true",
+        "endInclusive" -> "true"
+      )
+
+      InputFormatBase.addIterator(job, 
+        new IteratorSetting(2, "TimeColumnFilter", "org.apache.accumulo.core.iterators.user.ColumnSliceFilter", props))
+    }
+
+    InputFormatBase.fetchColumns(job, new APair(new Text(layerId.name), null: Text) :: Nil)
+  }
+
+  def reader(instance: AccumuloInstance, metaData: AccumuloLayerMetaData, keyBounds: KeyBounds[SpaceTimeKey])(implicit sc: SparkContext): FilterableRasterRDDReader[SpaceTimeKey] =
```

Comment:
This part feels pretty strange. I'm not sure I follow this entirely. Is there utility in returning a specifically typed object here? At the time this is instantiated/called both the metadata and LayerID are already on hand. Feels like it should just be a flat function/object with an interface specific to the package (hadoop/accumulo/etc).